### PR TITLE
User/britel/staging.new platforms

### PR DESCRIPTION
**Pull request by Britel**
This is a merge of master 01/07. We can delete the staging branch after the PR is approved.
The plan going forward to will keep new_platforms in line with master more regularly.

### CODE REVIEW

File: common/sgx/report.c [hunk 1]
```diff
@@ -190,3 +193,56 @@ oe_result_t oe_get_target_info(
...
+
+    if (!target_info_buffer || !target_info_size)
+    {
+        return OE_INVALID_PARAMETER;
+    }
+
+    *target_info_buffer = NULL;
+    *target_info_size = 0;
+
+    result = oe_get_target_info_v1(report, report_size, NULL, &temp_size);
```
Comment (dthaler) [inline common/sgx/report.c]: This is lame.   I thought our code did it the other way around... v2 is the clean code and v1 wraps v2.  That way when v1 is eventually deleted you don't have to rewrite the v2 code later, it's already well tested by then.

File: enclave/core/CMakeLists.txt [hunk 2]
```diff
@@ -1,96 +1,123 @@
...
+        list(APPEND PLATFORM_SRC
+            ../../common/sgx/rand.S
+            sgx/linux/reloc.c
+            sgx/linux/threadlocal.c)
+    elseif (WIN32)
+        list(APPEND PLATFORM_SRC
+            ../../common/sgx/rand.asm
+            sgx/windows/reloc.c)
+    endif()
+elseif(OE_TRUSTZONE)
```
Comment (dthaler) [inline enclave/core/CMakeLists.txt]: Shouldn't this be OE_USE_OPTEE not OE_TRUSTZONE?
After all, if you're using TZ with something other than OP-TEE, this section would be wrong

File: enclave/core/optee/bounds.c [hunk 3]
```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <openenclave/bits/defs.h>
+#include <openenclave/enclave.h>
+
+bool oe_is_within_enclave(const void* ptr, size_t sz)
```
Comment (dthaler) [inline enclave/core/optee/bounds.c]: Are these placeholders or what?  We have implementations of these for OP-TEE.

File: host/linux/windows.c [hunk 4]
```diff
@@ -20,15 +20,15 @@ HMODULE LoadLibraryExA(LPCSTR path, HANDLE file, DWORD flags)
         goto done;
 
     /* Allocate memory to hold the image (align on a page boundary). */
-    if (!(data = memalign(OE_PAGE_SIZE, st.st_size)))
+    if (!(data = memalign(OE_PAGE_SIZE, (size_t)st.st_size)))
```
Comment (dthaler) [inline host/linux/windows.c]: I guess I don't understand what this file is... linux/windows.c sounds like a contradiction

File: include/openenclave/enclave.h [hunk 5]
```diff
@@ -512,7 +576,8 @@ oe_result_t oe_verify_report(
  * Get a symmetric encryption key derived from the specified policy and coupled
  * to the enclave platform.
  *
- * @deprecated This function is deprecated. Use oe_get_seal_key_by_policy_v2() instead.
+ * @deprecated This function is deprecated. Use oe_get_seal_key_by_policy_v2()
+ instead.
```
Comment (dthaler) [inline include/openenclave/enclave.h]: revert line break or fix comment style

File: include/openenclave/enclave.h [hunk 6]
```diff
@@ -521,7 +586,8 @@ oe_result_t oe_verify_report(
  * small, this function sets it to the required size and returns
  * OE_BUFFER_TOO_SMALL. When this function success, the number of bytes written
  * to key_buffer is set to it.
- * @param key_info Optional buffer for the enclave-specific key information which
+ * @param key_info Optional buffer for the enclave-specific key information
+ which
```
Comment (dthaler) [inline include/openenclave/enclave.h]: revert line break or fix comment style

File: new_platforms/include/oehost.h [hunk 7]
```diff
@@ -24,7 +24,7 @@ extern "C" {
 typedef intptr_t ssize_t;
 #include <openenclave/bits/result.h>
 
-#ifndef __in_ecount
+#ifndef _MSC_VER
```
Comment (dthaler) [inline new_platforms/include/oehost.h]: oeenclave.h line 29 has instead:
#if !defined(_MSC_VER) || defined(OE_NO_SAL)
why are oehost.h and oeenclave.h different here?

File: common/sgx/report.c (hunk 1 above)
Comment (Britel) [reply common/sgx/report.c]: Agreed, new_platforms does have v1 calling v2 as far as I can tell. See oe_get_target_info_v2 in sgx_common.c. I don't think all the v2 versions have been implemented in the root definitions, we might want to bring this up before they are.

File: new_platforms/samples/sockets/host/server/main.c [hunk 8]
```diff
@@ -8,10 +8,10 @@
 #include <windows.h>
 #endif  // LINUX
 #include <openenclave/host.h>
-#include "SampleTA_u.h"
+#include "sockets_u.h"
 
 #ifdef OE_USE_SGX
-# define TA_ID "SampleTA" /* DLL will be SampleTA.signed.dll */
+# define TA_ID "sockets_enclave" /* DLL will be sockets_enclave.signed.dll */
```
[the remark this answers is not in the document]
Comment (Britel) [reply new_platforms/samples/sockets/host/server/main.c]: @HernanGatta

File: new_platforms/samples/sockets/host/client/main.c [hunk 9]
```diff
@@ -9,10 +9,10 @@
 #include <windows.h>
 #endif  // LINUX
 #include <openenclave/host.h>
-#include "SampleTA_u.h"
+#include "sockets_u.h"
 
 #ifdef OE_USE_SGX
-# define TA_ID "SampleTA" /* DLL will be SampleTA.signed.dll */
+# define TA_ID "sockets_enclave" /* DLL will be sockets_enclave.signed.dll */
```
[the remark this answers is not in the document]
Comment (Britel) [reply new_platforms/samples/sockets/host/client/main.c]: @HernanGatta

**Comment by Britel** [comment]
Will address comments in subsequent changes.

**Pull request merged**